File: ventura-sanic/middleware.py
from sanic.response import json

BRANDS = {
    'viventura': 1,
    'puraventura': 2,
    'japaventura': 3,
    'innoventura': 4,
}

def set_brand(app):
    """
    Validate/set the brands and locales per each request
    """
    @app.middleware('request')
    async def compress_request(request):
        project = request.headers.get('x-ventura-project', 'viventura')
        locale = request.headers.get('x-ventura-locale', 'de_DE')

        if not project or not locale:
            return json({'error':'x-ventura-* header is missing'}, 502)

        # set Locale id
        request.headers['locale_id'] = locale
        # set Project id
        request.headers['project_id'] = BRANDS[project]

File: ventura-express/middleware.js

const BRANDS = {
  'viventura': 1,
  'puraventura': 2,
  'japaventura': 3,
  'innoventura': 4,
}

module.exports = {
  /*
   * Validate/set the brands and locales per each request
   */
  setBrand: async function(req, res, next) {
    const project = req.get('x-ventura-project') || 'viventura';
    const locale = req.get('x-ventura-locale') || 'de_DE';

    if(!project || !locale) {
      res
        .status(502)
        .send({error: 'x-ventura-* header is missing'});
    }

    // set Locale id
    req.locale_id =  locale;
    // set Project id
    req.project_id = BRANDS[project]
    next();
  }
};
